rezz.lic

```diff
@@ -1,119 +1,126 @@
-#   Documentation: https://elanthipedia.play.net/Lich_script_repository#rezz
+=begin
+  Documentation: https://elanthipedia.play.net/Lich_script_repository#rezz
+=end
 
 custom_require.call(%w[common common-arcana events drinfomon spellmonitor])
 
-# rezz waggle set with POM if you want
-
 class Rezz
   include DRC
   include DRCA
-
+  
   def initialize
+    DRStats.cleric? ? nil : begin echo '*** Nice try, heathen! ***'; exit end
+	
     arg_definitions = [
       [
-        { name: 'person', regex: /\w+/, description: 'Person to rezz' }
+        { name: 'player', regex: /\w+/, optional: true, description: 'Name of the player to rezz' }
       ]
     ]
-
-    @rejuv_mana = 0
+	
     args = parse_args(arg_definitions)
     @settings = get_settings
-    rezz_prep(args.person)
+	
+	args.player ? @target = DRRoom.pcs.find { |name| /^#{args.player}/i =~ name }.to_a : find_dead
+	!@target ? begin echo "*** No bodies! ***"; exit end : nil
+	
+    @rezz = if @settings.waggle_sets['rezz']['Resurrection']
+              @settings.waggle_sets['rezz']['Resurrection']
+			else
+			  echo "Using default values for REZZ, make a 'rezz'=>Resurrection waggle set to avoid this."
+			  get_data('spells').spell_data['Resurrection']
+			end
+
+    @rejuv = if @settings.waggle_sets['rezz']['Rejuvenation']
+	           @settings.waggle_sets['rezz']['Rejuvenation']
+			 else
+			   echo "Using default values for REJUV, make a 'rezz'=>Rejuvenation waggle set to avoid this."
+	           get_data('spells').spell_data['Rejuvenation']
+			 end
+
+    Flags.add('no-favor', /has no favor with h.. god!/)
+    Flags.add('rejuv-silver', 'A thin silver nimbus surrounds', 'As you concentrate, a thin silver nimbus flickers into view around')
+    Flags.add('alive', 'Nothing happens')
+    Flags.add('necro', 'has no protection from the gods. judgment')
+
+    @target.each do |name|
```

Comment:
Is this ever going to be an array? Wouldn't it be simpler, because this isn't an array of multiple characters to raise, for this to just be put here without the `each do`? It's just confusing, but I could also just not be reading it correctly.

Reply:
Yes.  If you just run ;rezz, it will look at the people in the room and make an array of all the dead.

Reply:
Gotcha, thanks.

Reply:
"jayRyan24 3 days ago  Contributor
I also hope he gets froggy and makes it raise all dead in the room. :D But that's another pR. :P"

I got froggy.  ;rezz with no arguments will make an array of all the dead in the room.